roles/grub/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+---
+
+- name: Install grub tools
+  ansible.builtin.yum:
+    state: latest
+    update_cache: true
+    name: 'grub2-tools'
+  become: true
+
+- name: Check if /boot/default/grub exists
+  ansible.builtin.stat:
+    path: /etc/default/grub
+  register: defaultgrubpath
+  become: true
+
+- name: Set grub timeout in /etc/default/grub
+  ansible.builtin.replace:
+    path: /etc/default/grub
+    regexp: '^(GRUB_TIMEOUT)=[0-9]*$'
+    replace: '\1={{ grub_timeout_sec }}'
+  when: defaultgrubpath.stat.exists
+  become: true
+
+- name: Check if /boot/grub2/grub.cfg exists
+  ansible.builtin.stat:
+    path: /boot/grub2/grub.cfg
+  register: grub2path
+  become: true
+
+- name: Set grub timeout in /boot/grub2/grub.cfg
+  ansible.builtin.replace:
+    path: '/boot/grub2/grub.cfg'
+    regexp: '(set timeout)=[0-9]*'
+    replace: '\1={{ grub_timeout_sec }}'
+  when: grub2path.stat.exists
+  become: true
```

Comment:
Is this necessary? If I understand correctly, the change must be be made in `/etc/default/grub`. The `grub2-mkconfig --output /boot/grub2/grub.cfg` command will then update and overwrite `/boot/grub2/grub.cfg`. Manually patching the value here too is redundant.